### PR TITLE
Added tests for HttpRequestDataTableSource

### PR DESCRIPTION
**Pull request by bkonyi**

### CODE REVIEW

File: packages/devtools_app/lib/src/network/flutter/network_model.dart [hunk 1]
```diff
@@ -14,6 +14,9 @@ class HttpRequestDataTableSource extends DataTableSource {
     notifyListeners();
   }
 
+  @visibleForTesting
```
Comment (jacob314) [inline packages/devtools_app/lib/src/network/flutter/network_model.dart]: nit: provide a
```dart
@visibleForTesting
List<HttpRequestData> get data => _data
```
instead.

Comment (bkonyi) [reply packages/devtools_app/lib/src/network/flutter/network_model.dart]: Done.

File: packages/devtools_app/lib/src/network/network_controller.dart [hunk 2]
```diff
@@ -34,13 +34,12 @@ class NetworkController {
...
-    );
-    final outstandingRequestsMap = Map<String, HttpRequestData>.from(
-      _httpRequestsNotifier.value.outstandingRequests,
-    );
+  @visibleForTesting
+  static HttpRequests processHttpTimelineEventsHelper(
+      Timeline timeline,
+      int timelineMicrosOffset,
+      List<HttpRequestData> currentValues,
+      Map<String, HttpRequestData> outstandingRequestsMap) {
```
Comment (jacob314) [inline packages/devtools_app/lib/src/network/network_controller.dart]: nit: use flutter style by adding a trailing comma.

File: packages/devtools_app/lib/src/network/network_controller.dart [hunk 3]
```diff
@@ -95,13 +94,24 @@ class NetworkController {
...
       requests: currentValues,
       outstandingRequests: outstandingRequestsMap,
     );
   }
 
+  void processHttpTimelineEvents(Timeline timeline) {
+    final currentValues = List<HttpRequestData>.from(
+      _httpRequestsNotifier.value.requests,
+    );
+    final outstandingRequestsMap = Map<String, HttpRequestData>.from(
```
Comment (jacob314) [inline packages/devtools_app/lib/src/network/network_controller.dart]: nit: maybe inline these two variables.

File: packages/devtools_app/test/http_request_data_table_source_test.dart [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+    }
+
+    test('check defaults', () {
+      expect(dataTable.rowCount, 0);
+      expect(dataTable.isRowCountApproximate, false);
+      expect(dataTable.currentSelectionListenable.value, isNull);
+      expect(dataTable.selectedRowCount, 0);
+    });
+
+    test('verify utilities', () {
```
Comment (jacob314) [inline packages/devtools_app/test/http_request_data_table_source_test.dart]: nit: verify utilities -> time display

File: packages/devtools_app/test/http_request_data_table_source_test.dart [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+      // Check deselection works properly...
+      final deselectListener = () {
+        row = dataTable.getRow(0);
+        expect(row.selected, false);
+        expect(dataTable.selectedRowCount, 0);
+      };
+
+      // with clearSelection
+      dataTableListenerScope(
+          deselectListener, () => dataTable.clearSelection());
```
Comment (jacob314) [inline packages/devtools_app/test/http_request_data_table_source_test.dart]: nit: prefer flutter style.

**Pull request merged**